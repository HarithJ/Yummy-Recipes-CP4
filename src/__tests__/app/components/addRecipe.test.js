import React from 'react';
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson, { shallowToJson } from 'enzyme-to-json';

import AddRecipe from '../../../components/recipes/addRecipe.js'

describe("add recipe component", () => {
  const addRecipe = shallow(<AddRecipe />)

  it("renders properly", () => {
    expect(shallowToJson(addRecipe)).toMatchSnapshot();
  })

  it("has functions to facilitate adding of a recipe", () => {
    const preventDefault = jest.fn();

    expect(addRecipe.instance().handleAddRecipe({ preventDefault }));
    expect(addRecipe.instance().handleAddIng());
    expect(addRecipe.instance().clearInputs());
  })
})
