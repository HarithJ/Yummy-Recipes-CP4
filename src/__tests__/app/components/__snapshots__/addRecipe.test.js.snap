// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add recipe component renders properly 1`] = `
<div>
  <button
    className="btn btn-primary"
    data-target="#add-recipe-modal"
    data-toggle="modal"
    type="button"
  >
    Add Recipe
  </button>
  <div
    aria-hidden="true"
    aria-labelledby="addRecipeModal"
    className="modal fade"
    id="add-recipe-modal"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="modal-dialog modal-lg"
      role="document"
    >
      <div
        className="modal-content"
      >
        <form
          id="add-recipe-modal-form"
          name="addrecipeForm"
        >
          <div
            className="modal-header"
          >
            <input
              className="form-control recipetitle"
              name="recipetitle"
              onChange={[Function]}
              placeholder="Recipe Title"
              type="text"
              value=""
            />
            <button
              aria-label="Close"
              className="close"
              data-dismiss="modal"
              type="button"
            >
              <span
                aria-hidden="true"
              >
                Ã—
              </span>
            </button>
          </div>
          <div
            className="modal-body"
          >
            <h4>
              Ingredients
            </h4>
            <ul
              id="ingredients"
            />
            <button
              className="col-auto btn btn-primary"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fas fa-plus"
              />
            </button>
            <button
              className="col-auto btn btn-warning"
              onClick={[Function]}
              type="button"
            >
              clear
            </button>
            <h4>
              Directions
            </h4>
            <div
              className="row"
            >
              <div
                className="col-12"
              >
                <textarea
                  className="form-control directions"
                  name="directions"
                  onChange={[Function]}
                  rows="3"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="modal-footer"
          >
            <button
              className="btn btn-secondary"
              data-dismiss="modal"
              type="button"
            >
              Close
            </button>
            <button
              className="btn btn-primary"
              data-dismiss="modal"
              onClick={[Function]}
              type="submit"
            >
              Save changes
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
